### PR TITLE
Integrated ndx-icephys-meta schema with NWB core

### PR DESCRIPTION
**Pull request by oruebel**
This PR is for integration of the ndx-icephys-meta structures with NWB core. 

List of completed items:
* Declared SweepTable as deprecated  
* Declared icephys filtering field deprecatd
* Integrated the new types from the ndx-icephys-meta extensions schema 
   *  TimeSeriesReferenceVectorData
    * IntracellularRecordingsTable
    * SimultaneousRecordingsTable
    * SequentialRecordingsTable
    * RepetitionsTable
    * ExperimentalConditionsTable
* Added new icephys tables to NWBFile as part of /general/icephys

### CODE REVIEW

**Comment by oruebel** [comment]
@rly @ajtritt this PR is ready for review

File: core/nwb.icephys.yaml [hunk 1]
```diff
@@ -250,3 +253,170 @@ groups:
...
+        to fix the type of table that can be referenced here.
+  - name: sequential_recordings_index
+    neurodata_type_inc: VectorIndex
+    doc: Index dataset for the sequential_recordings column.
+
+- neurodata_type_def: ExperimentalConditionsTable
+  neurodata_type_inc: DynamicTable
+  name: experimental_conditions
+  doc: A table for grouping different intracellular recording repetitions together
+    that belong to the same experimental experimental_conditions.
```
Comment (oruebel) [inline core/nwb.icephys.yaml]:
```suggestion
    that belong to the same experimental condition.
```

File: core/nwb.file.yaml [hunk 2]
```diff
@@ -333,17 +333,60 @@ groups:
...
+          as a text attribute for each TimeSeries.'
         quantity: '?'
       groups:
       - neurodata_type_inc: IntracellularElectrode
         doc: An intracellular electrode.
         quantity: '*'
       - name: sweep_table
         neurodata_type_inc: SweepTable
-        doc: The table which groups different PatchClampSeries together.
+        doc: '[DEPRACATED] Table used to group different PatchClampSeries.SweepTable
```
Comment (rly) [inline core/nwb.file.yaml]:
```suggestion
        doc: '[DEPRECATED] Table used to group different PatchClampSeries. SweepTable
```

File: core/nwb.base.yaml [hunk 3]
```diff
@@ -3,6 +3,25 @@ datasets:
   neurodata_type_inc: Data
   doc: An abstract data type for a dataset.
 
+- neurodata_type_def: TimeSeriesReferenceVectorData
+  neurodata_type_inc: VectorData
+  dtype:
+  - name: idx_start
```
Comment (oruebel) [inline core/nwb.base.yaml]:
```suggestion
- neurodata_type_def: TimeSeriesReferenceVectorData
  neurodata_type_inc: VectorData
  default_name: timeseries
  dtype:
  - name: idx_start
```

File: core/nwb.icephys.yaml [hunk 4]
```diff
@@ -250,3 +253,170 @@ groups:
...
+    neurodata_type_inc: TimeSeriesReferenceVectorData
+    doc: Column storing the reference to the recorded response for the recording (rows)
+
+- neurodata_type_def: IntracellularRecordingsTable
+  neurodata_type_inc: AlignedDynamicTable
+  name: intracellular_recordings
+  doc: A table to group together a stimulus and response from a single electrode and
+    a single simultaneous recording. Each row in the table represents a single recording
+    consisting typically of a stimulus and a corresponding response. In some cases,
+    however, only a stimulus or a response are recorded as as part of an experiment.
```
Comment (rly) [inline core/nwb.icephys.yaml]:
```suggestion
    however, only a stimulus or a response is recorded as part of an experiment.
```

**Review by rly — APPROVED** [review]

**Comment by rly** [comment]
@oruebel Looks like tests in the accompanying https://github.com/NeurodataWithoutBorders/pynwb/pull/1349 pass so this is good to merge into dev.

**Pull request merged**